internal/providers/pluginfw/resources/notificationdestinations/data_notification_destinations.go

```diff
@@ -0,0 +1,126 @@
+package notificationdestinations
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/settings_tf"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func DataSourceNotificationDestinations() datasource.DataSource {
+	return &NotificationDestinationsDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &NotificationDestinationsDataSource{}
+
+type NotificationDestinationsDataSource struct {
+	Client *common.DatabricksClient
+}
+
+type NotificationDestinationsInfo struct {
+	Id                       types.String                                     `tfsdk:"id" tf:"computed"`
+	DisplayNameContains      types.String                                     `tfsdk:"display_name_contains" tf:"optional"`
+	Type                     types.String                                     `tfsdk:"type" tf:"optional"`
+	NotificationDestinations []settings_tf.ListNotificationDestinationsResult `tfsdk:"notification_destinations" tf:"computed"`
+}
+
+func (d *NotificationDestinationsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "databricks_notification_destinations_pluginframework"
```

Comment:
we can remove `_pluginframework` suffix, for other resources it's just part of the migration process